sql/tests: TestRandomSyntaxGeneration failed [nil pointer in createLogicalReplicationStreamPlanHook]

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18240630?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18240630?buildTab=artifacts#/) on master @ [3b6f3c659f7c511761b44d65ba2cf62815a514c1](https://github.com/cockroachdb/cockroach/commits/3b6f3c659f7c511761b44d65ba2cf62815a514c1):


```
=== RUN   TestRandomSyntaxGeneration
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration2500559134
    test_log_scope.go:76: use -show-logs to present logs inline
*
* ERROR: a panic has occurred!
* runtime error: invalid memory address or nil pointer dereference
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | runtime.panicmem
*   | 	GOROOT/src/runtime/panic.go:261
*   | runtime.sigpanic
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:328
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:138
*   | github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1
*   | 	pkg/sql/planhook.go:194
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	pkg/util/stop/stopper.go:498
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x5b763b2]

goroutine 58402 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x5b75d4a?, {0x89aa0c0, 0xc008c4fb00})
	pkg/util/stop/stopper.go:226 +0x65
panic({0x659c500?, 0xc995360?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects({0x89aa0c0, 0xc008c4fb90}, {0x8a69330, 0xc009642118}, {{0xc008a385c8, 0x1, 0x1}, {0x0, 0x0}}, 0x1)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:328 +0x532
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1({0x89aa0c0, 0xc008c4fb00}, {0x0?, 0x0?, 0x0?}, 0xc006c7cfc0)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:138 +0x5cf
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1({0x89aa0c0, 0xc008c4fb00})
	pkg/sql/planhook.go:194 +0x4a
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2({0x717d897?, 0x0?})
	pkg/util/stop/stopper.go:498 +0x1f0
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 57641
	pkg/util/stop/stopper.go:488 +0x47c
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #137481 sql/tests: TestRandomSyntaxGeneration failed [nil pointer dereference in backupPlanHook] [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-23.1.30-rc release-blocker]
- #132558 sql/tests: TestRandomSyntaxGeneration failed [CREATE TENANT statement is not valid after parsing then formatting] [C-test-failure O-robot P-2 T-db-server branch-release-24.1.6-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45725

Failing at this line: https://github.com/cockroachdb/cockroach/blob/f05cedf1958c08cfd8aef9ae5dcf1d208ae185b5/pkg/crosscluster/logical/create_logical_replication_stmt.go#L328

```
*
* ERROR: a panic has occurred!
* runtime error: invalid memory address or nil pointer dereference
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | runtime.panicmem
*   | 	GOROOT/src/runtime/panic.go:261
*   | runtime.sigpanic
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:328
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:138
*   | github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1
*   | 	pkg/sql/planhook.go:194
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	pkg/util/stop/stopper.go:498
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x5b763b2]

goroutine 58402 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x5b75d4a?, {0x89aa0c0, 0xc008c4fb00})
	pkg/util/stop/stopper.go:226 +0x65
panic({0x659c500?, 0xc995360?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects({0x89aa0c0, 0xc008c4fb90}, {0x8a69330, 0xc009642118}, {{0xc008a385c8, 0x1, 0x1}, {0x0, 0x0}}, 0x1)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:328 +0x532
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1({0x89aa0c0, 0xc008c4fb00}, {0x0?, 0x0?, 0x0?}, 0xc006c7cfc0)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:138 +0x5cf
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1({0x89aa0c0, 0xc008c4fb00})
	pkg/sql/planhook.go:194 +0x4a
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2({0x717d897?, 0x0?})
	pkg/util/stop/stopper.go:498 +0x1f0
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 57641
	pkg/util/stop/stopper.go:488 +0x47c
```

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18252780?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18252780?buildTab=artifacts#/) on master @ [951d890d1af51eefef402034c0506b5d6a9f014e](https://github.com/cockroachdb/cockroach/commits/951d890d1af51eefef402034c0506b5d6a9f014e):


```
=== RUN   TestRandomSyntaxGeneration
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration1025251692
    test_log_scope.go:76: use -show-logs to present logs inline
    rsg_test.go:937: 5s of 5m0s: 11630 executions, 646 successful
    rsg_test.go:937: 10s of 5m0s: 23929 executions, 1330 successful
    rsg_test.go:937: 15s of 5m0s: 35141 executions, 1970 successful
    rsg_test.go:937: 20s of 5m0s: 46442 executions, 2618 successful
*
* ERROR: a panic has occurred!
* runtime error: invalid memory address or nil pointer dereference
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | runtime.panicmem
*   | 	GOROOT/src/runtime/panic.go:261
*   | runtime.sigpanic
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:328
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:138
*   | github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1
*   | 	pkg/sql/planhook.go:194
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	pkg/util/stop/stopper.go:498
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x5b796b2]

goroutine 433342 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x5b7904a?, {0x89addc0, 0xc0202b8ab0})
	pkg/util/stop/stopper.go:226 +0x65
panic({0x659fba0?, 0xc99a360?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects({0x89addc0, 0xc0202b8b40}, {0x8a6d130, 0xc006f12118}, {{0xc0179fe330, 0x1, 0x1}, {0x0, 0x0}}, 0x1)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:328 +0x532
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1({0x89addc0, 0xc0202b8ab0}, {0x0?, 0x0?, 0x0?}, 0xc009d8f2c0)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:138 +0x5cf
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1({0x89addc0, 0xc0202b8ab0})
	pkg/sql/planhook.go:194 +0x4a
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2({0x7180a1f?, 0x0?})
	pkg/util/stop/stopper.go:498 +0x1f0
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 428295
	pkg/util/stop/stopper.go:488 +0x47c
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #137481 sql/tests: TestRandomSyntaxGeneration failed [nil pointer dereference in backupPlanHook] [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-23.1.30-rc release-blocker]
- #132558 sql/tests: TestRandomSyntaxGeneration failed [CREATE TENANT statement is not valid after parsing then formatting] [C-test-failure O-robot P-2 T-db-server branch-release-24.1.6-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18263982?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18263982?buildTab=artifacts#/) on master @ [0806ee44e36b62dce75175702b5bfb3db03e0577](https://github.com/cockroachdb/cockroach/commits/0806ee44e36b62dce75175702b5bfb3db03e0577):


```
    rsg_test.go:937: 20s of 5m0s: 46407 executions, 2152 successful
    rsg_test.go:937: 25s of 5m0s: 60805 executions, 2746 successful
    rsg_test.go:937: 30s of 5m0s: 75043 executions, 3334 successful
    rsg_test.go:937: 35s of 5m0s: 89095 executions, 3960 successful
    rsg_test.go:937: 40s of 5m0s: 102295 executions, 4481 successful
    rsg_test.go:937: 45s of 5m0s: 113379 executions, 4944 successful
    rsg_test.go:937: 50s of 5m0s: 122473 executions, 5307 successful
*
* ERROR: a panic has occurred!
* runtime error: invalid memory address or nil pointer dereference
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | runtime.panicmem
*   | 	GOROOT/src/runtime/panic.go:261
*   | runtime.sigpanic
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:328
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:138
*   | github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1
*   | 	pkg/sql/planhook.go:195
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	pkg/util/stop/stopper.go:498
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x5b8b912]

goroutine 800766 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x5b8b2aa?, {0x89ca800, 0xc010a802d0})
	pkg/util/stop/stopper.go:226 +0x65
panic({0x65b2240?, 0xc9d0360?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects({0x89ca800, 0xc010a80360}, {0x8a8a570, 0xc00b8a8698}, {{0xc0056b71b0, 0x1, 0x1}, {0x0, 0x0}}, 0x1)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:328 +0x532
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1({0x89ca800, 0xc010a802d0}, {0x0?, 0x0?, 0x0?}, 0xc015090c00)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:138 +0x5cf
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1({0x89ca800, 0xc010a802d0})
	pkg/sql/planhook.go:195 +0x4a
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2({0x7196787?, 0x0?})
	pkg/util/stop/stopper.go:498 +0x1f0
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 798552
	pkg/util/stop/stopper.go:488 +0x47c
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #137481 sql/tests: TestRandomSyntaxGeneration failed [nil pointer dereference in backupPlanHook] [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-23.1.30-rc release-blocker]
- #132558 sql/tests: TestRandomSyntaxGeneration failed [CREATE TENANT statement is not valid after parsing then formatting] [C-test-failure O-robot P-2 T-db-server branch-release-24.1.6-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18268953?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18268953?buildTab=artifacts#/) on master @ [01f9e61532862cbbdbc64180013ca1cb57f4a017](https://github.com/cockroachdb/cockroach/commits/01f9e61532862cbbdbc64180013ca1cb57f4a017):


```
    rsg_test.go:937: 15s of 5m0s: 39530 executions, 2113 successful
    rsg_test.go:937: 20s of 5m0s: 51503 executions, 2633 successful
    rsg_test.go:937: 25s of 5m0s: 63890 executions, 3164 successful
    rsg_test.go:937: 30s of 5m0s: 76059 executions, 3649 successful
    rsg_test.go:937: 35s of 5m0s: 86712 executions, 4096 successful
    rsg_test.go:937: 40s of 5m0s: 98804 executions, 4596 successful
    rsg_test.go:937: 45s of 5m0s: 109683 executions, 5086 successful
*
* ERROR: a panic has occurred!
* runtime error: invalid memory address or nil pointer dereference
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | runtime.panicmem
*   | 	GOROOT/src/runtime/panic.go:261
*   | runtime.sigpanic
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:328
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:138
*   | github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1
*   | 	pkg/sql/planhook.go:195
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	pkg/util/stop/stopper.go:498
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x5b8bb72]

goroutine 670932 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x5b8b50a?, {0x89caba0, 0xc02997d4d0})
	pkg/util/stop/stopper.go:226 +0x65
panic({0x65b24c0?, 0xc9d0360?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects({0x89caba0, 0xc02a2f8360}, {0x8a8a910, 0xc025d11098}, {{0xc0274165c0, 0x1, 0x1}, {0x0, 0x0}}, 0x1)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:328 +0x532
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1({0x89caba0, 0xc02997d4d0}, {0x0?, 0x0?, 0x0?}, 0xc0290087e0)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:138 +0x5cf
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1({0x89caba0, 0xc02997d4d0})
	pkg/sql/planhook.go:195 +0x4a
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2({0x7196a07?, 0x0?})
	pkg/util/stop/stopper.go:498 +0x1f0
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 670231
	pkg/util/stop/stopper.go:488 +0x47c
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #137481 sql/tests: TestRandomSyntaxGeneration failed [nil pointer dereference in backupPlanHook] [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-23.1.30-rc release-blocker]
- #132558 sql/tests: TestRandomSyntaxGeneration failed [CREATE TENANT statement is not valid after parsing then formatting] [C-test-failure O-robot P-2 T-db-server branch-release-24.1.6-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18285273?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18285273?buildTab=artifacts#/) on master @ [d58f071217505b226a097d39f87072f4e1bd8e06](https://github.com/cockroachdb/cockroach/commits/d58f071217505b226a097d39f87072f4e1bd8e06):


```
    rsg_test.go:937: 10s of 5m0s: 24171 executions, 1371 successful
    rsg_test.go:937: 15s of 5m0s: 37130 executions, 2104 successful
    rsg_test.go:937: 20s of 5m0s: 49474 executions, 2795 successful
    rsg_test.go:937: 25s of 5m0s: 61996 executions, 3444 successful
    rsg_test.go:937: 30s of 5m0s: 72337 executions, 4007 successful
    rsg_test.go:937: 35s of 5m0s: 83805 executions, 4620 successful
    rsg_test.go:937: 40s of 5m0s: 94153 executions, 5097 successful
*
* ERROR: a panic has occurred!
* runtime error: invalid memory address or nil pointer dereference
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | runtime.panicmem
*   | 	GOROOT/src/runtime/panic.go:261
*   | runtime.sigpanic
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:328
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:138
*   | github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1
*   | 	pkg/sql/planhook.go:187
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	pkg/util/stop/stopper.go:498
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x5b8c1b2]

goroutine 586199 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x5b8bb4a?, {0x89cb6a0, 0xc01a213ae0})
	pkg/util/stop/stopper.go:226 +0x65
panic({0x65b2e40?, 0xc9d6360?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects({0x89cb6a0, 0xc01a213b80}, {0x8a8b410, 0xc020e20b18}, {{0xc009f4d8f8, 0x1, 0x1}, {0x0, 0x0}}, 0x1)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:328 +0x532
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1({0x89cb6a0, 0xc01a213ae0}, 0xc01d4f8720)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:138 +0x5cf
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1({0x89cb6a0, 0xc01a213ae0})
	pkg/sql/planhook.go:187 +0x3d
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2({0x7197a4d?, 0x0?})
	pkg/util/stop/stopper.go:498 +0x1f0
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 584407
	pkg/util/stop/stopper.go:488 +0x47c
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #137481 sql/tests: TestRandomSyntaxGeneration failed [nil pointer dereference in backupPlanHook] [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-23.1.30-rc release-blocker]
- #132558 sql/tests: TestRandomSyntaxGeneration failed [CREATE TENANT statement is not valid after parsing then formatting] [C-test-failure O-robot P-2 T-db-server branch-release-24.1.6-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18299190?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18299190?buildTab=artifacts#/) on master @ [270ba1e8fe656fab0a182643fb77a6a5be64c1b0](https://github.com/cockroachdb/cockroach/commits/270ba1e8fe656fab0a182643fb77a6a5be64c1b0):


```
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration23666326
    test_log_scope.go:76: use -show-logs to present logs inline
    rsg_test.go:937: 5s of 5m0s: 9803 executions, 534 successful
    rsg_test.go:937: 10s of 5m0s: 21164 executions, 1110 successful
    rsg_test.go:937: 15s of 5m0s: 35439 executions, 1764 successful
    rsg_test.go:937: 20s of 5m0s: 47213 executions, 2274 successful
    rsg_test.go:937: 25s of 5m0s: 59023 executions, 2739 successful
*
* ERROR: a panic has occurred!
* runtime error: invalid memory address or nil pointer dereference
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | runtime.panicmem
*   | 	GOROOT/src/runtime/panic.go:261
*   | runtime.sigpanic
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:328
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:138
*   | github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1
*   | 	pkg/sql/planhook.go:187
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	pkg/util/stop/stopper.go:498
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x5b8c1b2]

goroutine 422752 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x5b8bb4a?, {0x89cb6a0, 0xc0031e7680})
	pkg/util/stop/stopper.go:226 +0x65
panic({0x65b2e40?, 0xc9d6360?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects({0x89cb6a0, 0xc0031e7720}, {0x8a8b410, 0xc00b819098}, {{0xc019547ce0, 0x1, 0x1}, {0x0, 0x0}}, 0x1)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:328 +0x532
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1({0x89cb6a0, 0xc0031e7680}, 0xc016cbf860)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:138 +0x5cf
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1({0x89cb6a0, 0xc0031e7680})
	pkg/sql/planhook.go:187 +0x3d
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2({0x7197a4d?, 0x0?})
	pkg/util/stop/stopper.go:498 +0x1f0
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 421275
	pkg/util/stop/stopper.go:488 +0x47c
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #137481 sql/tests: TestRandomSyntaxGeneration failed [nil pointer dereference in backupPlanHook] [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-23.1.30-rc release-blocker]
- #132558 sql/tests: TestRandomSyntaxGeneration failed [CREATE TENANT statement is not valid after parsing then formatting] [C-test-failure O-robot P-2 T-db-server branch-release-24.1.6-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18308230?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18308230?buildTab=artifacts#/) on master @ [a0faa4e779ac9e54d17e8400141b0aa472887974](https://github.com/cockroachdb/cockroach/commits/a0faa4e779ac9e54d17e8400141b0aa472887974):


```
    rsg_test.go:937: 5s of 5m0s: 10276 executions, 543 successful
    rsg_test.go:937: 10s of 5m0s: 19299 executions, 969 successful
    rsg_test.go:937: 15s of 5m0s: 26930 executions, 1342 successful
    rsg_test.go:937: 20s of 5m0s: 34386 executions, 1675 successful
    rsg_test.go:937: 25s of 5m0s: 44288 executions, 2062 successful
    rsg_test.go:937: 30s of 5m0s: 54621 executions, 2531 successful
    rsg_test.go:937: 35s of 5m0s: 64927 executions, 2947 successful
*
* ERROR: a panic has occurred!
* runtime error: invalid memory address or nil pointer dereference
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | runtime.panicmem
*   | 	GOROOT/src/runtime/panic.go:261
*   | runtime.sigpanic
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:328
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:138
*   | github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1
*   | 	pkg/sql/planhook.go:187
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	pkg/util/stop/stopper.go:498
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x5b8c332]

goroutine 516366 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x5b8bcca?, {0x89cb8e0, 0xc0245cf2c0})
	pkg/util/stop/stopper.go:226 +0x65
panic({0x65b3040?, 0xc9d6360?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects({0x89cb8e0, 0xc0245cf360}, {0x8a8b650, 0xc013d00698}, {{0xc012157e30, 0x1, 0x1}, {0x0, 0x0}}, 0x1)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:328 +0x532
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1({0x89cb8e0, 0xc0245cf2c0}, 0xc02427f1a0)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:138 +0x5cf
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1({0x89cb8e0, 0xc0245cf2c0})
	pkg/sql/planhook.go:187 +0x3d
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2({0x7197c4d?, 0x0?})
	pkg/util/stop/stopper.go:498 +0x1f0
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 512839
	pkg/util/stop/stopper.go:488 +0x47c
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #137481 sql/tests: TestRandomSyntaxGeneration failed [nil pointer dereference in backupPlanHook] [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-23.1.30-rc release-blocker]
- #132558 sql/tests: TestRandomSyntaxGeneration failed [CREATE TENANT statement is not valid after parsing then formatting] [C-test-failure O-robot P-2 T-db-server branch-release-24.1.6-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18316417?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18316417?buildTab=artifacts#/) on master @ [686a8fa2cbd2193542949d7d4069725fa36db18d](https://github.com/cockroachdb/cockroach/commits/686a8fa2cbd2193542949d7d4069725fa36db18d):


```
    rsg_test.go:937: 25s of 5m0s: 61275 executions, 2712 successful
    rsg_test.go:937: 30s of 5m0s: 74899 executions, 3293 successful
    rsg_test.go:937: 35s of 5m0s: 88043 executions, 3876 successful
    rsg_test.go:937: 40s of 5m0s: 101435 executions, 4464 successful
    rsg_test.go:937: 45s of 5m0s: 112413 executions, 4911 successful
    rsg_test.go:937: 50s of 5m0s: 120628 executions, 5232 successful
    rsg_test.go:937: 55s of 5m0s: 131802 executions, 5694 successful
*
* ERROR: a panic has occurred!
* runtime error: invalid memory address or nil pointer dereference
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | runtime.panicmem
*   | 	GOROOT/src/runtime/panic.go:261
*   | runtime.sigpanic
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:328
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:138
*   | github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1
*   | 	pkg/sql/planhook.go:187
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	pkg/util/stop/stopper.go:498
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x5b8c0f2]

goroutine 758338 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x5b8ba8a?, {0x89cb6a0, 0xc005a7efa0})
	pkg/util/stop/stopper.go:226 +0x65
panic({0x65b2e00?, 0xc9d6360?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects({0x89cb6a0, 0xc005a7f040}, {0x8a8b410, 0xc00d76b618}, {{0xc029cbcaf8, 0x1, 0x1}, {0x0, 0x0}}, 0x1)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:328 +0x532
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1({0x89cb6a0, 0xc005a7efa0}, 0xc00e27a4e0)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:138 +0x5cf
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1({0x89cb6a0, 0xc005a7efa0})
	pkg/sql/planhook.go:187 +0x3d
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2({0x7197a0d?, 0x0?})
	pkg/util/stop/stopper.go:498 +0x1f0
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 757784
	pkg/util/stop/stopper.go:488 +0x47c
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #137481 sql/tests: TestRandomSyntaxGeneration failed [nil pointer dereference in backupPlanHook] [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-23.1.30-rc release-blocker]
- #132558 sql/tests: TestRandomSyntaxGeneration failed [CREATE TENANT statement is not valid after parsing then formatting] [C-test-failure O-robot P-2 T-db-server branch-release-24.1.6-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18323366?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/18323366?buildTab=artifacts#/) on master @ [f9df57e2bebd963d10ffd7fa52e4d37cf01b80df](https://github.com/cockroachdb/cockroach/commits/f9df57e2bebd963d10ffd7fa52e4d37cf01b80df):


```
    rsg_test.go:937: 15s of 5m0s: 13286 executions, 741 successful
    rsg_test.go:937: 20s of 5m0s: 18325 executions, 1009 successful
    rsg_test.go:937: 25s of 5m0s: 22485 executions, 1276 successful
    rsg_test.go:937: 30s of 5m0s: 26676 executions, 1498 successful
    rsg_test.go:937: 35s of 5m0s: 30969 executions, 1707 successful
    rsg_test.go:937: 40s of 5m0s: 35243 executions, 1952 successful
    rsg_test.go:937: 45s of 5m0s: 39331 executions, 2187 successful
*
* ERROR: a panic has occurred!
* runtime error: invalid memory address or nil pointer dereference
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | runtime.panicmem
*   | 	GOROOT/src/runtime/panic.go:261
*   | runtime.sigpanic
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:328
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1
*   | 	pkg/crosscluster/logical/create_logical_replication_stmt.go:138
*   | github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1
*   | 	pkg/sql/planhook.go:187
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	pkg/util/stop/stopper.go:498
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x5b8e772]

goroutine 211292 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x5b8e10a?, {0x89ce580, 0xc00c4b3900})
	pkg/util/stop/stopper.go:226 +0x65
panic({0x65b5480?, 0xc9da360?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.resolveDestinationObjects({0x89ce580, 0xc00c4b39a0}, {0x8a8e2f0, 0xc00e9ac118}, {{0xc002988cf0, 0x1, 0x1}, {0x0, 0x0}}, 0x1)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:328 +0x532
github.com/cockroachdb/cockroach/pkg/crosscluster/logical.createLogicalReplicationStreamPlanHook.func1({0x89ce580, 0xc00c4b3900}, 0xc011035e00)
	pkg/crosscluster/logical/create_logical_replication_stmt.go:138 +0x5cf
github.com/cockroachdb/cockroach/pkg/sql.(*hookFnNode).startExec.func1({0x89ce580, 0xc00c4b3900})
	pkg/sql/planhook.go:187 +0x3d
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2({0x719a3ef?, 0x0?})
	pkg/util/stop/stopper.go:498 +0x1f0
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 196448
	pkg/util/stop/stopper.go:488 +0x47c
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #137481 sql/tests: TestRandomSyntaxGeneration failed [nil pointer dereference in backupPlanHook] [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-23.1.30-rc release-blocker]
- #132558 sql/tests: TestRandomSyntaxGeneration failed [CREATE TENANT statement is not valid after parsing then formatting] [C-test-failure O-robot P-2 T-db-server branch-release-24.1.6-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/crosscluster/logical/BUILD.bazel
+++ b/pkg/crosscluster/logical/BUILD.bazel
@@ -100,6 +100,7 @@ go_library(
 go_test(
     name = "logical_test",
     srcs = [
+        "create_logical_replication_stmt_test.go",
         "dead_letter_queue_test.go",
         "logical_replication_job_test.go",
         "lww_kv_processor_test.go",
@@ -138,6 +139,7 @@ go_test(
         "//pkg/sql/catalog/descpb",
         "//pkg/sql/catalog/descs",
         "//pkg/sql/catalog/desctestutils",
+        "//pkg/sql/catalog/resolver",
         "//pkg/sql/execinfra",
         "//pkg/sql/execinfrapb",
         "//pkg/sql/isql",
@@ -166,6 +168,7 @@ go_test(
         "//pkg/util/uuid",
         "@com_github_cockroachdb_cockroach_go_v2//crdb",
         "@com_github_cockroachdb_errors//:errors",
+        "@com_github_cockroachdb_redact//:redact",
         "@com_github_lib_pq//:pq",
         "@com_github_stretchr_testify//require",
     ],

--- a/pkg/crosscluster/logical/create_logical_replication_stmt.go
+++ b/pkg/crosscluster/logical/create_logical_replication_stmt.go
@@ -38,6 +38,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/asof"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/catid"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
 	"github.com/cockroachdb/cockroach/pkg/sql/syntheticprivilege"
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
 	"github.com/cockroachdb/cockroach/pkg/util/buildutil"
@@ -135,7 +136,7 @@ func createLogicalReplicationStreamPlanHook(
 				return pgerror.Newf(pgcode.InvalidParameterValue, "unknown discard option %q", m)
 			}
 		}
-		resolvedDestObjects, err := resolveDestinationObjects(ctx, p, stmt.Into, stmt.CreateTable)
+		resolvedDestObjects, err := resolveDestinationObjects(ctx, p, p.SessionData(), stmt.Into, stmt.CreateTable)
 		if err != nil {
 			return err
 		}
@@ -305,7 +306,8 @@ func (r *ResolvedDestObjects) TargetDescription() string {
 
 func resolveDestinationObjects(
 	ctx context.Context,
-	p sql.PlanHookState,
+	r resolver.SchemaResolver,
+	sessionData *sessiondata.SessionData,
 	destResources tree.LogicalReplicationResources,
 	createTable bool,
 ) (ResolvedDestObjects, error) {
@@ -315,15 +317,16 @@ func resolveDestinationObjects(
 		if err != nil {
 			return resolved, err
 		}
-		dstObjName.HasExplicitSchema()
-
+		dstTableName := dstObjName.ToTableName()
 		if createTable {
-			_, _, resPrefix, err := resolver.ResolveTarget(ctx,
-				&dstObjName, p, p.SessionData().Database, p.SessionData().SearchPath)
+			found, _, resPrefix, err := resolver.ResolveTarget(ctx,
+				&dstObjName, r, sessionData.Database, sessionData.SearchPath)
 			if err != nil {
 				return resolved, errors.Newf("resolving target import name")
 			}
-
+			if !found {
+				return resolved, errors.Newf("database or schema not found for destination table %s", destResources.Tables[i])
+			}
 			if resolved.ParentDatabaseID == 0 {
 				resolved.ParentDatabaseID = resPrefix.Database.GetID()
 				resolved.ParentSchemaID = resPrefix.Schema.GetID()
@@ -332,7 +335,9 @@ func resolveDestinationObjects(
 			} else if resolved.ParentSchemaID != resPrefix.Schema.GetID() {
 				return resolved, errors.Newf("destination tables must all be in the same schema")
 			}
-
+			if _, _, err := resolver.ResolveMutableExistingTableObject(ctx, r, &dstTableName, true, tree.ResolveRequireTableDesc); err == nil {
+				return resolved, errors.Newf("destination table %s already exists", destResources.Tables[i])
+			}
 			tbNameWithSchema := tree.MakeTableNameWithSchema(
 				tree.Name(resPrefix.Database.GetName()),
 				tree.Name(resPrefix.Schema.GetName()),
@@ -340,10 +345,9 @@ func resolveDestinationObjects(
 			)
 			resolved.TableNames = append(resolved.TableNames, tbNameWithSchema)
 		} else {
-			dstTableName := dstObjName.ToTableName()
-			prefix, td, err := resolver.ResolveMutableExistingTableObject(ctx, p, &dstTableName, true, tree.ResolveRequireTableDesc)
+			prefix, td, err := resolver.ResolveMutableExistingTableObject(ctx, r, &dstTableName, true, tree.ResolveRequireTableDesc)
 			if err != nil {
-				return resolved, err
+				return resolved, errors.Wrapf(err, "failed to find existing destination table %s", destResources.Tables[i])
 			}
 
 			tbNameWithSchema := tree.MakeTableNameWithSchema(

--- a/pkg/crosscluster/logical/create_logical_replication_stmt_test.go
+++ b/pkg/crosscluster/logical/create_logical_replication_stmt_test.go
@@ -1,0 +1,202 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package logical
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/security/username"
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descs"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/resolver"
+	"github.com/cockroachdb/cockroach/pkg/sql/isql"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
+	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/redact"
+	"github.com/stretchr/testify/require"
+)
+
+func TestResolveDestinationObjects(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	skip.UnderDeadlock(t)
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+
+	srv, conn, _ := serverutils.StartServer(t, base.TestServerArgs{})
+	defer srv.Stopper().Stop(ctx)
+	s := srv.ApplicationLayer()
+
+	execCfg := s.ExecutorConfig().(sql.ExecutorConfig)
+
+	sqlDB := sqlutils.MakeSQLRunner(conn)
+	sqlDB.Exec(t, "CREATE DATABASE db1")
+	sqlDB.Exec(t, "CREATE SCHEMA db1.sc1")
+	sqlDB.Exec(t, "CREATE SCHEMA db1.sc2")
+	sqlDB.Exec(t, "CREATE TABLE db1.sc1.t1 (a INT PRIMARY KEY)")
+	sqlDB.Exec(t, "CREATE TABLE db1.sc1.t2 (a INT PRIMARY KEY)")
+	sqlDB.Exec(t, "CREATE TABLE t3 (a INT PRIMARY KEY)")
+
+	sqlUser, err := username.MakeSQLUsernameFromUserInput("root", username.PurposeValidation)
+	require.NoError(t, err)
+
+	resolveObjects := func(destResources tree.LogicalReplicationResources, createTables bool) (resolved ResolvedDestObjects, err error) {
+		err = sql.TestingDescsTxn(ctx, s, func(ctx context.Context, txn isql.Txn, _ *descs.Collection) error {
+			opName := redact.SafeString("resolve")
+			sessionData := sql.NewInternalSessionData(ctx, execCfg.Settings, opName)
+			sessionData.Database = "defaultdb"
+			planner, close := sql.NewInternalPlanner(
+				opName,
+				txn.KV(),
+				sqlUser,
+				&sql.MemoryMetrics{},
+				&execCfg,
+				sessionData,
+			)
+			defer close()
+			resolved, err = resolveDestinationObjects(ctx, planner.(resolver.SchemaResolver), sessionData, destResources, createTables)
+			return err
+		})
+		return resolved, err
+	}
+
+	res := func(db string, tables ...string) tree.LogicalReplicationResources {
+		resources := tree.LogicalReplicationResources{
+			Database: tree.Name(db),
+		}
+		for _, table := range tables {
+			t := strings.Split(table, ".")
+			resources.Tables = append(resources.Tables, tree.NewUnresolvedName(t...))
+		}
+		return resources
+	}
+
+	type testCase struct {
+		name         string
+		resources    tree.LogicalReplicationResources
+		create       bool
+		expectedDesc string
+		expectedErr  string
+	}
+
+	for _, tc := range []testCase{
+		{
+			name:         "single",
+			resources:    res("", "db1.sc1.t1"),
+			expectedDesc: "db1.sc1.t1",
+		},
+		{
+			name:         "single/create",
+			resources:    res("", "db1.sc1.t1_c"),
+			create:       true,
+			expectedDesc: "db1.sc1.t1_c",
+		},
+		{
+			name:         "implicit_schema",
+			resources:    res("", "defaultdb.t3"),
+			expectedDesc: "defaultdb.public.t3",
+		},
+		{
+			name:         "implicit_schema/create",
+			resources:    res("", "defaultdb.t3_c"),
+			create:       true,
+			expectedDesc: "defaultdb.public.t3_c",
+		},
+		{
+			name:         "implicit_db",
+			resources:    res("", "t3"),
+			expectedDesc: "defaultdb.public.t3",
+		},
+		{
+			name:         "implicit_db/create",
+			resources:    res("", "t3_c"),
+			create:       true,
+			expectedDesc: "defaultdb.public.t3_c",
+		},
+		{
+			name:         "multi",
+			resources:    res("", "db1.sc1.t1", "db1.sc1.t2"),
+			expectedDesc: "db1.sc1.t1, db1.sc1.t2",
+		},
+		{
+			name:         "multi/create",
+			resources:    res("", "db1.sc1.t1_c", "db1.sc1.t2_c"),
+			create:       true,
+			expectedDesc: "db1.sc1.t1_c, db1.sc1.t2_c",
+		},
+		{
+			name:        "missing_schema",
+			resources:   res("", "db1.s2.t1"),
+			expectedErr: "failed to find existing destination table db1.s2.t1",
+		},
+		{
+			name:        "missing_db",
+			resources:   res("", "db2.t1"),
+			expectedErr: "failed to find existing destination table db2.t1",
+		},
+		{
+			name:        "missing_schema/create",
+			resources:   res("", "db1.s2.t1"),
+			create:      true,
+			expectedErr: "database or schema not found for destination table db1.s2.t1",
+		},
+		{
+			name:        "missing_db/create",
+			resources:   res("", "db2.t1"),
+			create:      true,
+			expectedErr: "database or schema not found for destination table db2.t1",
+		},
+		{
+			name:         "multiple_db_target",
+			resources:    res("", "t3", "db1.sc1.t1"),
+			expectedDesc: "defaultdb.public.t3, db1.sc1.t1",
+		},
+		{
+			name:        "multiple_db_target/create",
+			resources:   res("", "t3_c", "db1.sc1.t1_c"),
+			create:      true,
+			expectedErr: "destination tables must all be in the same database",
+		},
+		{
+			name:        "multiple_schema_target/create",
+			resources:   res("", "db1.sc2.t2_c", "db1.sc1.t1_c"),
+			create:      true,
+			expectedErr: "destination tables must all be in the same schema",
+		},
+		{
+			name:        "existing_table/create",
+			resources:   res("", "db1.sc1.t1"),
+			create:      true,
+			expectedErr: "destination table db1.sc1.t1 already exists",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			resolved, err := resolveObjects(tc.resources, tc.create)
+			if tc.expectedErr != "" {
+				require.ErrorContains(t, err, tc.expectedErr)
+				return
+			}
+			require.NoError(t, err)
+			require.Equal(t, tc.expectedDesc, resolved.TargetDescription())
+			if tc.create {
+				require.NotZero(t, resolved.ParentDatabaseID)
+				require.NotZero(t, resolved.ParentSchemaID)
+			} else {
+				require.Zero(t, resolved.ParentDatabaseID)
+				require.Zero(t, resolved.ParentSchemaID)
+			}
+		})
+
+	}
+
+}

